azavea/sentinel-2-stac-utils
# Improve error tolerance

## Overview

This PR substantially improves handling for two classes of errors:

- when we don't get json we can decode
- when s3 values are missing

It also does a few other quality of life things:

- uses the Sentinel-2 collection to decide what the collection id should be
- makes item links have a rel type of item instead of child (#20)
- makes logs quieter / configurable with an environment variable (#19)

### Checklist

- ~New tests have been added or existing tests have been modified~

### Testing Instructions

- `git checkout bf42663`
- Kill all the lines in `short-product-info-l2a.csv` except the first few
- Create a catalog: `AWS_PROFILE=raster-foundry AWS_REGION=us-east-1 S2STAC_LOG_LEVEL=INFO bloop run s2stac -- create --collection L2A --inventory-path $(pwd)/data/short-product-info-l2a.csv --output-catalog-root wherever/you/want`
- logs should be a lot quieter, everything should be fine but you should retry sometimes when you get unlucky / lucky, however you want to think about it
- in either or both of the lines in the csv, add some random letters to the path
- run the import again
- you should get a nice message about what S3 is mad about and a log at the end of the run about which paths were bad
- remove your random letters and instead switch a `productInfo.json` to a `tileInfo.json`
- run the import again
- you should get a nice message about the json decoding failure (unless you rolled a lucky s3 error, in which case you should see retries until you get a json decoding failure) and a log at the end of the run about which paths were bad

Closes #18 (if applicable)

Closes #19 

Closes #20
